pkg/executor/simple.go

```diff
@@ -2522,14 +2518,13 @@ func (e *SimpleExec) executeSetPwd(ctx context.Context, s *ast.SetPwdStmt) error
 		h = s.User.Hostname
 
 		checker := privilege.GetPrivilegeManager(e.Ctx())
-		checker.MatchIdentity(ctx, u, h, false)
```

Comment:
Just curious about what this line is used for  before this PR

Reply:
In the past, `ensureActiveUser()` is called for both 'show grants' and 'show grants for user xxx'
This change make `ensureActiveUser` only called for the later case.